AsyncDisplayKit/ASViewController.m

```diff
@@ -0,0 +1,47 @@
+//
+//  ASViewController.m
+//  AsyncDisplayKit
+//
+//  Created by Huy Nguyen on 16/09/15.
+//  Copyright (c) 2015 Facebook. All rights reserved.
+//
+
+#import "ASViewController.h"
+#import "ASAssert.h"
+#import "ASDimension.h"
+
+@implementation ASViewController
+
+- (instancetype)initWithNode:(ASDisplayNode *)node
+{
+  if (!(self = [super init])) {
+    return nil;
+  }
+  
+  ASDisplayNodeAssertNotNil(node, @"Node must not be nil");
+  ASDisplayNodeAssertTrue(!node.layerBacked);
+  _node = node;
+  
+  return self;
+}
+
+- (void)loadView
+{
+  ASDisplayNodeAssertTrue(!_node.layerBacked);
+  self.view = _node.view;
+}
+
+- (void)viewWillLayoutSubviews
+{
+  ASSizeRange constrainedSize = ASSizeRangeMake(self.view.bounds.size, self.view.bounds.size);
+  [_node measureWithSizeRange:constrainedSize];
```

Comment:
Right — it's totally fine for this to be synchronous.  Later, we can add the API to allow a user to trigger preparation of the view in advance, which would include layout.  My question is - what if a user dispatch_async'd a call to measure:, and then pushed the view controller?

The answer may be that we get bad behavior.  That's also not a big deal right now, as it would be unusual to manually call measure: in that way and then push the view — but it will be a key part of the final version of 2.0 to support kicking off asynchronous layout, and then having a call that is able to block on the remainder of it (e.g. "-ensureMeasurementComplete" or something — that is probably not a good name).  It would return immediately if done, otherwise wait on the background operation.